### PR TITLE
Create DCF_Demo.md

### PR DESCRIPTION
**Pull request by cindydeng1998**
added some description and expected outcome from the Azure CLI of running this demo

### CODE REVIEW

File: DCF_Demo.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+// expected outcome: serial terminal output
+Receive direct method: dm.1.install
+	Payload: {"package_name":"cipher_v1i1"}
+Create producer for cipher v1i1...
+Interface cipher 1 published with success
+```
+
+We are now sendind a message to the device and using the newly installed cipher to encrypt the message "Welcome to Azure IoT!". The response will be the encrypted result of the message.
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "cipher.1.encrypt" --mp "{\"context\":0, \"src\":\"Welcome to Azure IoT\!\"}"
```
Comment (mamokarz) [inline DCF_Demo.md:105]: >az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "cipher.1.encrypt" --mp "{\"context\":0, \"src\":\"Welcome to Azure IoT\!\"}" [](http://example.com/codeflow?start=0&length=154)

It does not work in powershell, you must use \`" for double quotes.   #WontFix

Comment (mamokarz) [reply DCF_Demo.md:105]:
```
az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "cipher.1.encrypt" --mp "{\`"context\`":0, \`"src\`":\`"Welcome to Azure IoT!\`"}"
```

File: DCF_Demo.md [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+Install sprinker interface
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "dm.1.install" --mp "{\"package_name\":\"sprinkler_v1i1\"}" 
+
+// expected outcome
+{
+  "payload": {},
+  "status": 200
+}
+```
```
Comment (mamokarz) [inline DCF_Demo.md:43]: Let's add a ipc_query here to show that the new interface sprinkler.1 is now part of the available interfaces.  #Resolved

File: DCF_Demo.md [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+      "dm.1",
+      "sprinkler.1",
+      "cipher.1"
+    ]
+  },
+  "status": 200
+}
+```
+
+Now you can try and use a new context of the cipher interface, which would then call a different cryptography algorithm that the cipher has. However, the new context has not been installed yet.
```
Comment (mamokarz) [inline DCF_Demo.md]: >has not been installed yet [](http://example.com/codeflow?start=166&length=26)

Consider "the new context is not supported by the version 1 of the cipher, it requires an update to the version 2." #Resolved

File: DCF_Demo.md [hunk 4]
```diff
@@ -0,0 +1,513 @@
...
+
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "cipher.1.encrypt" --mp "{\"context\":1, \"src\":\"Welcome to Azure IoT\!\"}" 
+
+// expected outcome
+{
+  "payload": {
+    "description": "AZ_ERROR_ITEM_NOT_FOUND"
```
Comment (mamokarz) [inline DCF_Demo.md]: >AZ_ERROR_ITEM_NOT_FOUND [](http://example.com/codeflow?start=20&length=23)

AZ_ERROR_NOT_SUPPORTED #Resolved

File: DCF_Demo.md [hunk 5]
```diff
@@ -0,0 +1,513 @@
...
+  },
+  "status": 404
```
Comment (mamokarz) [inline DCF_Demo.md]: >404 [](http://example.com/codeflow?start=12&length=3)

405 #Resolved

File: DCF_Demo.md [hunk 6]
```diff
@@ -0,0 +1,513 @@
...
+
+Install sprinker interface
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "dm.1.install" --mp "{\`"package_name\`":\`"sprinkler_v1i1\`"}"  
+
+// expected outcome
+{
+  "payload": {},
+  "status": 200
+}
```
Comment (mamokarz) [inline DCF_Demo.md:318]: Consider to have only an example of the \`" instead of repeat the entire sequence.  #WontFix

File: DCF_Demo.md [hunk 7]
```diff
@@ -0,0 +1,513 @@
+# DCF Invoke Device Method Using Azure CLI
+
+## Open up an instance of Azure CLI 
+You can use the Azure Portal or an instance on Powershell/bash/other local environments
```
Comment (smcd253) [inline DCF_Demo.md:7]: let's remove PS for now bc it does not follow the same format #Resolved

Comment (mamokarz) [reply DCF_Demo.md:7]: Why not keep it and add a note to replace \" by \`" in PS?

**Review by smcd253 — APPROVED** [review]
lgtm

File: DCF_Demo.md (hunk 7 above)
Comment (smcd253) [reply DCF_Demo.md:7]: I think we should keep it simple. too many approaches will only make it more confusing than it needs to be

Comment (cindydeng1998) [reply DCF_Demo.md:7]: I added a click to expand section so people can just open the one they want if that works, I personally don't have azcli installed on powershell and used the portal for convenience

File: DCF_Demo.md (hunk 2 above)
Comment (cindydeng1998) [reply DCF_Demo.md:43]: added in c5d48459b23391e198b444a6831f5917a546909c

File: DCF_Demo.md (hunk 3 above)
Comment (cindydeng1998) [reply DCF_Demo.md]: fixed wording

File: DCF_Demo.md (hunk 4 above)
Comment (cindydeng1998) [reply DCF_Demo.md]: fixed in c5d48459b23391e198b444a6831f5917a546909c

File: DCF_Demo.md (hunk 5 above)
Comment (cindydeng1998) [reply DCF_Demo.md]: fixed in c5d48459b23391e198b444a6831f5917a546909c

File: DCF_Demo.md [hunk 8]
```diff
@@ -0,0 +1,550 @@
...
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "ipc_query.1.query" --mp "{}" 
+
+// expected outcome : Azure CLI
+{
+  "payload": {
+    "continuation_token": 655615,
+    "result": [
+      "ipc_query.1",
+      "dm.1"
```
Comment (mamokarz) [inline DCF_Demo.md]: "sprinkler.1" should still be part of this list #Resolved

File: DCF_Demo.md [hunk 9]
```diff
@@ -0,0 +1,550 @@
...
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "ipc_query.1.query" --mp "{}" 
+
+// expected outcome
+{
+  "payload": {
+    "continuation_token": 655615,
+    "result": [
+      "ipc_query.1",
+      "dm.1",
+      "cipher.1"
```
Comment (mamokarz) [inline DCF_Demo.md:228]: Hear too, add "sprinkler.1" #Resolved

File: DCF_Demo.md [hunk 10]
```diff
@@ -0,0 +1,550 @@
...
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "ipc_query.1.query" --mp "{}"
+
+// expected outcome
+{
+  "payload": {
+    "continuation_token": 655615,
+    "result": [
+      "ipc_query.1",
+      "dm.1"
```
Comment (mamokarz) [inline DCF_Demo.md]: Please add "sprinkler.1" #Resolved

File: DCF_Demo.md [hunk 11]
```diff
@@ -0,0 +1,550 @@
...
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "ipc_query.1.query" --mp "{}" 
+
+// expected outcome : Azure CLI
+{
+  "payload": {
+    "continuation_token": 655615,
+    "result": [
+      "ipc_query.1",
+      "dm.1"
```
Comment (mamokarz) [inline DCF_Demo.md]: Please add "sprinkler.1" #Resolved

File: DCF_Demo.md [hunk 12]
```diff
@@ -0,0 +1,550 @@
...
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "ipc_query.1.query" --mp "{}" 
+
+// expected outcome
+{
+  "payload": {
+    "continuation_token": 655615,
+    "result": [
+      "ipc_query.1",
+      "dm.1",
+      "cipher.1"
```
Comment (mamokarz) [inline DCF_Demo.md:504]: Please add "sprinkler.1" #Resolved

**Review by mamokarz — APPROVED** [review]
:shipit:

File: DCF_Demo.md [hunk 13]
```diff
@@ -0,0 +1,554 @@
+# DCF Invoke Device Method Using Azure CLI
+
+## Open up an instance of Azure CLI
```
Comment (smcd253) [inline DCF_Demo.md:6]: There should be a section here describing the steps one should follow before they can start this demo. (i.e. following the stm getting started guide)

**Pull request merged**